apps/cbportal/base/CBGrid/.macros/page/0accessnode/3_0accessnode.py

```diff
@@ -0,0 +1,62 @@
+import requests
+import json
+import datetime
+from jose import jwt as jose_jwt
+import ast
+
+def main(j, args, params, tags, tasklet):
+    page = args.page
+    counter = """
+<script>
+window.onload=function(){
+var counter = %s;
+var interval = setInterval(function() {
+    counter--;
+    document.getElementById('counter').innerText = "The following data will be expired in " + counter + " seconds if you didn't use it"
+    if (counter <= 0) {
+        document.getElementById('counter').innerText = "this data is no longer valid"
+    }
+}, 1000);
+}
+</script>
+    """
+    remote = args.requestContext.params.get('ip')
+    jwt = args.requestContext.env['beaker.session'].get('jwt')
+    if jwt:
+        jwt_data = jose_jwt.get_unverified_claims(jwt)
+        jwt_data = ast.literal_eval(jwt_data)
+        exp_date = datetime.datetime.fromtimestamp(jwt_data["exp"])
+        now = datetime.datetime.now()
+        if now > exp_date:
+            url = 'https://itsyou.online/v1/oauth/jwt/refresh'
+            headers = {'Authorization': 'bearer {0}'.format(jwt)}
+            resp = requests.get(url, headers=headers)
+            jwt = ""
+            if resp.status_code == 200:
+                jwt = resp.content
```

Comment:
need to set the new jwt in the session, by this u will end up requesting jwt per every request